File: qe-selenium-python/Activity_Thirteen.py
from selenium import webdriver
from selenium.webdriver.common.by import By

driver = webdriver.Firefox()

driver.get("https://training-support.net/webelements/tables")
driver.maximize_window()

print(driver.title)

rows = driver.find_elements(By.XPATH, "//table//tr")
print("Number of rows:", len(rows))

columns = driver.find_elements(By.XPATH, "//table//tr[1]/td")
print("Number of columns:", len(columns))

print("Values in the third row:")
third_row_cells = driver.find_elements(By.XPATH, "//table//tr[3]/td")

for cell in third_row_cells:
    print(cell.text)

value = driver.find_element(By.XPATH, "//table//tr[2]//td[2]").text
print("Value at 2nd Row, 2nd Column:", value)

driver.quit()

File: qe-selenium-java/src/main/java/demos/ActivityThirteen.java
package demos;

import java.util.*;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class ActivityThirteen {

    public static void main(String[] args) throws InterruptedException {
        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/tables");
        driver.manage().window().maximize();

        System.out.println(driver.getTitle());

        List<WebElement> rows = driver.findElements(By.xpath("//table//tr"));
        System.out.println("Number of rows: " + rows.size());

        List<WebElement> columns = driver.findElements(By.xpath("//table//tr[1]/td"));
        System.out.println("Number of columns: " + columns.size());

        System.out.println("Values in the third row: ");
        List<WebElement> thirdRowCells = driver.findElements(By.xpath("//table//tr[3]/td"));

        for (WebElement cell : thirdRowCells)
            System.out.println(cell.getText());

        System.out.println(
                "Value at 2nd Row, 2nd Column: " + driver.findElement(By.xpath("//table//tr[2]//td[2]")).getText());

        driver.quit();
    }

}
